### DISCUSSION TITLE
Server-to-server scalable architecture

### DISCUSSION THREAD

**Discussion by snz2** [uNetworking/uWebSockets.js#255] [Q&A] [answered]
Hi I am trying to implement a real time service.
I will use server-to-server communication with aws autoscaling + loadbalancer.

So the most commonly used
I am going to synchronize messages between servers with multiple worker and pubsub, but I want to simplify it a bit.

uWebsocket.js recently supported pubsub, but I know it needs a client for server-to-server synchronization (though I'm not sure if I'm wrong)

Currently, it is used as pubsub embedded in other key value db.
To simplify this architecture further, I want to use uws as a broker (not mq), and npm ws seems to be the most suitable for clients.

In short

[  uws pubsub broker  ]
                         
    messsage sync with pubsub
              
[ uws / ws ] [ uws / ws ] [ ... ] [ ... ]

I'm not sure yet
Whether to use npm ws or red ... (key value db)
 Is there a better solution within uWebsocket?

**Comment by ghost** [comment, marked as answer]
You can use Redis pub/sub between servers and then uWS pub/sub as the browser-facing pub/sub. Don't overuse Redis though, I've seen companies with this set-up where they had 1-to-1 subscription (one Redis subscription for every browser client). That had horrible performance. That's my two cents.
